content/sensu-go/5.17/reference/secrets.md

```diff
@@ -0,0 +1,214 @@
+description  | The name for the secret that is used internally by Sensu.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"name": "sensu-ansible-token"{{< /highlight >}}
+
+namespace    |      |
+-------------|------
+description  | [Sensu RBAC namespace][9] that the secret belongs to.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"namespace": "default"{{< /highlight >}}
+
+### Spec attributes
+
+id           | 
+-------------|------ 
+description  | The identifying key for the provider to retrieve the secret.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"id": "secret/ansible#token"{{< /highlight >}}
+
+provider     | 
+-------------|------ 
+description  | The name of the Sensu provider with the secret.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"provider": "ansible_vault"{{< /highlight >}}
+
+## Secret configuration
+
+Use the [Secrets API][2] and [sensuctl][3] to create, view, and manage your secrets configuration.
+To manage secrets configuration with sensuctl, configure sensuctl as the default [`admin` user][6].
+
+The [standard sensuctl subcommands][4] are available for secrets (list, info, and delete).
+
+To list all secrets:
+
+{{< highlight shell >}}
+sensuctl secret list
+{{< /highlight >}}
+
+To see a secret's status:
+
+{{< highlight shell >}}
+sensuctl secret info SECRET_NAME
+{{< /highlight >}}
+
+To delete a secret:
+
+{{< highlight shell >}}
+sensuctl secret delete SECRET_NAME
+{{< /highlight >}}
+
+## Secret examples
+
+A secret resource definition refers to a secrets `id` and a secrets `provider`.
+Read the [secrets provider reference][7] for the provider specification.
+
+{{< language-toggle >}}
+
+{{< highlight yml >}}
+---
+type: Secret
+api_version: secrets/v1
+metadata:
+  name: sensu-ansible-token
+  namespace: default
+spec:
+  id: ANSIBLE_TOKEN
+  provider: env
+{{< /highlight >}}
+
+{{< highlight json >}}
+{
+  "type": "Secret",
+  "api_version": "secrets/v1",
+  "metadata": {
+    "name": "sensu-ansible-token",
+    "namespace": "default"
+  },
+  "spec": {
+    "id": "ANSIBLE_TOKEN",
+    "provider": "env"
+  }
+}
+{{< /highlight >}}
+
+{{< /language-toggle >}}
+
+Secrets that target a HashiCorp Vault always have an `id` as shown in the following example:
+
+{{< language-toggle >}}
+
+{{< highlight yml >}}
+---
+type: Secret
+api_version: secrets/v1
+metadata:
+  name: sensu-ansible-token
+  namespace: default
+spec:
+  id: 'secret/website#database'
+  provider: ansible_vault
+{{< /highlight >}}
+
+{{< highlight json >}}
+{
+  "type": "Secret",
+  "api_version": "secrets/v1",
+  "metadata": {
+    "name": "database",
+    "namespace": "default"
+  },
+  "spec": {
+    "id": "secret/website#database",
+    "provider": "vault"
+  }
+}
+{{< /highlight >}}
+
+{{< /language-toggle >}}
+
+Secrets that target a HashiCorp Vault must start with the word `secret`.
```

Comment:
I'm not sure this is true, I thought this depends on the keyspace that a Vault operator stores their secrets in. @echlebek @palourde wdyt?

Reply:
You're right. From [Vault docs](https://www.vaultproject.io/api-docs/secret/kv/kv-v2/):

> This documentation assumes the kv secrets engine is enabled at the /secret path in Vault. Since it is possible to enable secrets engines at any location, please update your API calls accordingly.

Reply:
I started to change it below, but Eric mentioned it when we were working on the guide earlier. Here's what the guide says:

"For secrets that target a HashiCorp Vault, the `id` must start with the word `secret/`."

Is that incorrect?

Reply:
Let me elaborate.

The path can be arbitrary, like the vault documentation says. The thing is, the `secret` keyspace is already set up with policies, at least in the dev server, and I would imagine the production Vault server as well.

For that reason I'd like to steer users towards using this structure. Users that know better can use a different id scheme, and users that don't understand vault won't get tripped up in needing to do additional work.

Perhaps we need to tweak the language to capture this nuance rather than just saying "it must be so"?

Reply:
@echlebek Thank you for clarifying.

I suggest changing: "Secrets that target a HashiCorp Vault must start with the word `secret`."

To:
"The `id` value for secrets that target a HashiCorp Vault must start with the name of the secret's path in Vault.
The [Vault dev server][10] is preconfigured with the `secret` keyspace already set up.
This is convenient for learning and getting started with Vault secrets management, so this example and our guide to [Secrets management with HashiCorp Vault][11] use the `secret/` path for the `id` value.

Reply:
Thank you!